src/main/java/core/basesyntax/OperationWithFruits.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import core.basesyntax.exeptions.NotEnoughFruitsException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class OperationWithFruits {
+
+    private RemoveFruit buyingOperation = new RemoveFruit();
+    private AddFruit addOperation = new AddFruit();
+
+    public List<Fruit> operationWithFruits(List<Transaction> transaction)
+            throws NotEnoughFruitsException {
+        List<Fruit> fruitsAvailable = new ArrayList<>();
+        for (int i = 0; i < transaction.size(); i++) {
+            String operationType = transaction.get(i).getTypeOfOperation();
+            switch (operationType) {
+                case "s":
+                    fruitsAvailable = addOperation
+                            .operation(fruitsAvailable, transaction.get(i));
+                    break;
+                case "b":
+                    fruitsAvailable = buyingOperation
+                            .operation(fruitsAvailable, transaction.get(i));
+                    break;
+                case "r":
+                    fruitsAvailable = addOperation
+                            .operation(fruitsAvailable, transaction.get(i));
+                    break;
+                default:
+                    break;
```

Comment:
this can be replaced with a `Map<String, Operaton>`
based on the string you receive the operator and call its `operate()` method
(I would recommend to name it `apply()`)

Reply:
done